jwt/algorithms.py

```diff
@@ -96,6 +103,17 @@ def prepare_key(self, key):
         if isinstance(key, text_type):
             key = key.encode('utf-8')
 
+        invalid_strings = [
+            b'-----BEGIN PUBLIC KEY-----',
+            b'-----BEGIN CERTIFICATE-----',
+            b'ssh-rsa'
```

Comment:
I'm aware this PR has already been merged, but I wanted to open a brief discussion.

Another possible header is `-----BEGIN RSA PUBLIC KEY-----`, and there may be others.  Is there a more sound way of dealing with this that's somewhat more future-proof?  If the check looked like this, would it be too broad?
```
invalid_strings = [
            b'-----',
            b'ssh-rsa'
]
```